### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by blackjackkent**
Information about branching strategy as discussed in Gitter; skeleton requirements for issue submission, PRs and commit messages.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Contributing to Pipeline
+
+Want to go with the flow? Follow the guidelines below to contribute to the project:
```
Comment (bvtsang) [inline CONTRIBUTING.md]: If you want this to rhyme, you can say "Want to go with the flow? Follow the guidelines below!"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#### Table Of Contents
+
+* [Submitting Issues](#submitting-issues)
+* [Branching Strategy](#branching-strategy)
+* [Pull Requests](#pull-requests)
+* [Git Commit Messages](#git-commit-messages)
+
+## Submitting Issues
+
+* If you have read the guidelines below, submit an issue at https://github.com/new-xkit/pipeline/issues
```
Comment (bvtsang) [inline CONTRIBUTING.md]: I would rephrase this as "If you want to report an issue, first please read the guidelines below, then submit an issue at https://github.com/new-xkit/pipeline/issues".

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+* Include screenshots where possible, as well as exact steps to reproduce the bug
+
+## Branching Strategy
+
+* The `production` branch always represents code which is user-facing -- i.e. the most recently deployed production code.
+* The `master` branch is the central hub for all development work.
+* `release-candidate` branches (formatted e.g. `rc-1.1.0`) are branched from master in preparation for a production deployment; they serve as pre-production snapshots suitable for final bugfixes and release.
+** When a `release-candidate` branch receives final approval, it is merged into both `master` and `production` and then deleted.
```
Comment (bvtsang) [inline CONTRIBUTING.md]: This should probably be formatted as:

```markdown
  * When a `release-candidate` branch receives final approval, it is merged into both `master` and `production` and then deleted.
```

Note the spaces in front.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+* `feature` branches are branched from `master` when a developer begins work on a feature. All work for that feature should be isolated to that branch; it is then submitted as a pull request back to `master` when it is ready for review and merging.
```
Comment (bvtsang) [inline CONTRIBUTING.md]: Nitpick: move this line before the one about the `release-candidate` branch, only because it's probably more relevant to the average contributor.

**Comment by bvtsang** [comment]
Just to clarify, are hotfixes made to the `production` branch, and then merged back into `master`? Or do they just follow the standard `release-candidate` process?

**Comment by Wolvan** [comment]
yes, @bvtsang, hotfixes branch off from production and merge into release and production at the end

**Comment by bvtsang** [comment]
Do you mean "branch off from `master`"?

**Comment by Wolvan** [comment]
@bvtsang branch off of master, yes

**Comment by bvtsang** [comment]
@Wolvan Thanks for the clarification.

Under the heading for commit messages, I think we should also include something on explaining the what and the why instead of the how (which should be explained in the code). Specifically, I'm talking about [this piece of advice](http://chris.beams.io/posts/git-commit/#why-not-how).

File: CONTRIBUTING.md (hunk 1 above)
Comment (blackjackkent) [inline CONTRIBUTING.md]: I deliberately avoided that tbh. :P

**Comment by bvtsang** [comment]
LGTM
